### PR TITLE
Add taxpasta taxdir check

### LINKED ISSUE #390: Taxpasta parameters produce an error exit status of 2 which asks for a taxonomy even though kraken2 and centrifuge have worked.
### Description of the bug

Hi everyone, I'm trying to add taxpasta parameters (e.g. --taxpasta_add_name, --taxpasta_add_rank) but I keep getting a command error exit status of 2 with the error being that a taxonomy is required. I asked on Slack and @jfy133 has suggested that it may be something to do with Taxonomy nodes/names dump files? Thanks!

### Command used and terminal output

```console
Command: 

nextflow run nf-core/taxprofiler -r 1.1.0 -profile docker --perform_shortread_qc --shortread_qc_tool fastp --shortread_qc_adapter1 AGATCGGAAGAGCACACGTCTGAACTCCAGTCA --shortread_qc_adapter2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT --perform_shortread_hostremoval --save_hostremoval_unmapped --shortread_hostremoval_index /data/Ali/databases/GRCh38_noalt_as_Phix --hostremoval_reference /data/Ali/databases/GRCh38_noalt_as_Phix/GCA_002596845.1_ASM259684v1_genomic.fna.gz --input /data/Ali/sample_lists/taxprofiler.s001script.D1.csv --databases /data/Ali/database_lists/taxprofiler.kraken2.centrifuge.krakenuniq.kaiju.csv --outdir /data/Ali/output/tax.2031.out --run_kraken2 --run_bracken --run_centrifuge --run_krakenuniq --run_profile_standardisation --run_krona --taxpasta_add_name --taxpasta_add_rank --taxpasta_add_lineage --taxpasta_add_idlineage --taxpasta_add_ranklineage

Error output:

[Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'NFCORE_TAXPROFILER:TAXPROFILER:STANDARDISATION_PROFILES:TAXPASTA_STANDARDISE (kraken2|kraken2db)'

Caused by:
  Process `NFCORE_TAXPROFILER:TAXPROFILER:STANDARDISATION_PROFILES:TAXPASTA_STANDARDISE (kraken2|kraken2db)` terminated with an error exit status (2)

Command executed:

  taxpasta standardise \
      -p kraken2 -o kraken2_kraken2db.tsv --add-name --add-rank --add-lineage --add-id-lineage \
       \
      D1_se_D1_kraken2db.kraken2.kraken2.report.txt
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_TAXPROFILER:TAXPROFILER:STANDARDISATION_PROFILES:TAXPASTA_STANDARDISE":
      taxpasta: $(taxpasta --version)
  END_VERSIONS

Command exit status:
  2

Command output:
  [17:47:09] CRITICAL The '--add-name' option requires a        standardise.py:192
                      taxonomy. Please provide one using the                      
                      option '--taxonomy'.                                        

Command error:
  [17:47:09] CRITICAL The '--add-name' option requires a        standardise.py:192
                      taxonomy. Please provide one using the                      
                      option '--taxonomy'.
```

### Relevant files

_No response_

### System information

Taxprofiler version 1.1.0
Docker
Linux

### PR DESCRIPTION
**Pull request by jfy133**
closes #390 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - [#379](https://github.com/nf-core/taxprofiler/pull/379) Added support for previously missing Bracken-corrected Kraken2 report as output (added by @hkaspersen & @jfy133 )
 - [#380](https://github.com/nf-core/taxprofiler/pull/380) Updated to nf-core pipeline template v2.10 (added by @LilyAnderssonLee & @sofstam)
+- [#393](https://github.com/nf-core/taxprofiler/pull/383) Add validation check for a taxpasta taxonomy directory if --taxpasta*add*\* parameters requested (♥️ to alimalrashed for reporting, added by @jfy133)
```
Comment (jfy133) [inline CHANGELOG.md:12]:
```suggestion
- [#393](https://github.com/nf-core/taxprofiler/pull/383) Add validation check for a taxpasta taxonomy directory if --taxpasta_add_* parameters requested (♥️ to alimalrashed for reporting, added by @jfy133)
```

File: workflows/taxprofiler.nf [hunk 2]
```diff
@@ -53,6 +53,8 @@ if (params.diamond_save_reads              ) log.warn "[nf-core/taxprofiler] DIA
 if (params.run_malt && params.run_krona && !params.krona_taxonomy_directory) log.warn "[nf-core/taxprofiler] Krona can only be run on MALT output if path to Krona taxonomy database supplied to --krona_taxonomy_directory. Krona will not be executed in this run for MALT."
 if (params.run_bracken && !params.run_kraken2) error('ERROR: [nf-core/taxprofiler] You are attempting to run Bracken without running kraken2. This is not possible! Please set --run_kraken2 as well.')
 
+if ( (params.taxpasta_add_name || params.taxpasta_add_rank || params.taxpasta_add_lineage || params.taxpasta_add_lineage || params.taxpasta_add_idlineage || params.taxpasta_add_ranklineage ) && !params.taxpasta_taxonomy_dir ) error('ERROR: [nf-core/taxprofiler] All --taxpasta_add_* parameters require a taxonomy supplied to --taxpasta_taxonomy_dir. However the latter parameter was not detected. Please check input.')
```
Comment (Midnighter) [inline workflows/taxprofiler.nf:56]: Easier to understand in my opinion:

```suggestion
if ( [params.taxpasta_add_name, params.taxpasta_add_rank, params.taxpasta_add_lineage, params.taxpasta_add_lineage, params.taxpasta_add_idlineage, params.taxpasta_add_ranklineage].any() && !params.taxpasta_taxonomy_dir ) error('ERROR: [nf-core/taxprofiler] All --taxpasta_add_* parameters require a taxonomy supplied to --taxpasta_taxonomy_dir. However the latter parameter was not detected. Please check input.')
```

**Review by Midnighter — APPROVED** [review]
I didn't check why the tests are failing, the code looks good to me.

File: workflows/taxprofiler.nf (hunk 2 above)
Comment (jfy133) [reply workflows/taxprofiler.nf:56]: Yup makes sensE!

**Comment by jfy133** [comment]
Final failures are problem with GHA. running locally fine :+1:

**Pull request merged**